app/src/hooks/initialize-agent.ts

```diff
@@ -0,0 +1,262 @@
+import { getBCAgentModules } from '../helpers/bc-agent-modules'
+import { BCState, BCLocalStorageKeys } from '../store'
+
+const loadCachedLedgers = async (): Promise<IndyVdrPoolConfig[] | undefined> => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const cachedTransactions = await PersistentStorage.fetchValueForKey<any>(BCLocalStorageKeys.GenesisTransactions)
+  if (cachedTransactions) {
+    const { timestamp, transactions } = cachedTransactions
+    return moment().diff(moment(timestamp), 'days') >= 1 ? undefined : transactions
+  }
+}
+
+const useInitializeBCAgent = () => {
+  const { agent, setAgent } = useAgent()
+  const [store, dispatch] = useStore<BCState>()
+  const { walletSecret } = useAuth()
+  const [logger, indyLedgers, attestationMonitor, credDefs, schemas] = useServices([
+    TOKENS.UTIL_LOGGER,
+    TOKENS.UTIL_LEDGERS,
+    TOKENS.UTIL_ATTESTATION_MONITOR,
+    TOKENS.CACHE_CRED_DEFS,
+    TOKENS.CACHE_SCHEMAS,
+  ])
+
+  const restartExistingAgent = useCallback(async () => {
+    if (!walletSecret?.id || !walletSecret.key || !agent) {
+      return
+    }
+
+    logger.info('Agent already initialized, restarting...')
+
+    try {
+      await agent.wallet.open({
+        id: walletSecret.id,
+        key: walletSecret.key!,
+      })
+    } catch (error) {
+      // Credo does not use error codes but this will be in the
+      // the error message if the wallet is already open
+      const catchPhrase = 'instance already opened'
+
+      if (error instanceof WalletError && error.message.includes(catchPhrase)) {
+        logger.warn('Wallet already open, nothing to do')
+      } else {
+        logger.error('Error opening existing wallet:', error as Error)
+
+        throw new BifoldError(
+          'Wallet Service',
+          'There was a problem unlocking the wallet.',
+          (error as Error).message,
+          1047
+        )
+      }
+
+      await agent.mediationRecipient.initiateMessagePickup()
+    }
+  }, [walletSecret, agent, logger])
+
+  const createNewAgent = useCallback(
+    async (ledgers: IndyVdrPoolConfig[]): Promise<Agent | undefined> => {
+      if (!walletSecret?.id || !walletSecret.key) {
+        return
+      }
+
+      logger.info('No agent initialized, creating a new one')
+
+      const options = {
+        config: {
+          label: store.preferences.walletName || 'BC Wallet',
+          walletConfig: {
+            id: walletSecret.id,
+            key: walletSecret.key,
+          },
+          logger,
+          mediatorPickupStrategy: MediatorPickupStrategy.Implicit,
+          autoUpdateStorageOnStartup: true,
+          autoAcceptConnections: true,
+        },
+        dependencies: agentDependencies,
+        modules: getBCAgentModules({
+          indyNetworks: ledgers,
+          mediatorInvitationUrl: Config.MEDIATOR_URL,
+          txnCache: {
+            capacity: 1000,
+            expiryOffsetMs: 1000 * 60 * 60 * 24 * 7,
```

Comment:
would it make sense to keep this math in a const file?
nit: might also be nice to have a comment like: // 7 days in milliseconds

Reply:
I'd probably put it in a const file if it were used in more than one file. If it's just local to this maybe define it at the top because it's used in two different places in the same file.

Reply:
This was copied exactly as is from the splash screen in the spirit of keeping all the logic the same but I can make this change anyway